### ISSUE TITLE
lifetime may not live long enough

### ISSUE THREAD

**Issue by jennydaman** [dcchut/async-recursion#37]
Hello,

I am wondering how to deal with this situation.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9c9d1ddc0b3ece4b4bb40280c0fda6ca

```rust
use async_recursion::async_recursion;

#[tokio::main]
async fn main() {
    count_down(5, None).await;
}

#[async_recursion]
async fn count_down(num: u32, foo: Option<&str>) {
    if num == 0 {
        return;
    }
    println!("{}, {:?}", num, foo);
    count_down(num - 1, foo);
}
```

Output:

```
error: lifetime may not live long enough
 --> src/main.rs:8:1
  |
8 | #[async_recursion]
  | ^^^^^^^^^^^^^^^^^^ returning this value requires that `'1` must outlive `'static`
9 | async fn count_down(num: u32, foo: Option<&str>) {
  |                                           - let's call the lifetime of this reference `'1`
  |
  = note: this error originates in the attribute macro `async_recursion` (in Nightly builds, run with -Z macro-backtrace for more info)
help: to declare that the trait object captures data from argument `foo`, you can add an explicit `'_` lifetime bound
  |
8 | #[async_recursion] + '_
  |                    ++++
```

The "help" given by rustc is invalid.

**Comment by jennydaman** [comment]
If I were to expand the macro and accept the compiler's help, it works:

```rust
#[must_use]
fn count_down(
    num: u32,
    foo: Option<&str>,
) -> ::core::pin::Pin<Box<dyn ::core::future::Future<Output = ()> + ::core::marker::Send + '_>> {
    Box::pin(async move {
        if num == 0 {
            return;
        }
        println!("{}, {:?}", num, foo);
        count_down(num - 1, foo);
    })
}
```

**Comment by dcchut** [comment]
Looks like a duplicate of #34

**Comment by dcchut** [comment]
Resolved in #38, released in v1.1.0.
